SPECS/reaper/reaper_build_caches.sh

```diff
@@ -0,0 +1,168 @@
+#! /bin/bash
+
+set -euo pipefail
+
+# NOTE: EXECUTING with id=0 (root) CAUSES REAPER BUILD TO FAIL.
+if [[ $(id -u) -eq 0 ]]; then
+    echo "Must run script as non-root user!" >&2
+    exit 1
+fi
+
+# Cassandra reaper version for which to generate build caches.
+# (a) Modify VERSION="x.x.x" in script.
+# (b) Pass arguments to command line: ./reaper_build_caches.sh 3.1.1
+VERSION="3.1.1"
+if [ $# -ge 1 ]; then
+    VERSION=$1
+fi
+
+echo "Building Cassandra cache for version '$VERSION'."
+
+# URL to download the sources to build
+SOURCE_URL="https://github.com/thelastpickle/cassandra-reaper/archive/refs/tags/${VERSION}.tar.gz"
+
+# Build cache names
+BOWER_COMPONENTS="reaper-bower-components-${VERSION}.tar.gz"
+SRC_UI_NODE_MODULES="reaper-srcui-node-modules-${VERSION}.tar.gz"
+BOWER_CACHE="reaper-bower-cache-${VERSION}.tar.gz"
+MAVEN_CACHE="reaper-m2-cache-${VERSION}.tar.gz"
+NPM_CACHE="reaper-npm-cache-${VERSION}.tar.gz"
+LOCAL_LIB_NODE_MODULES="reaper-local-lib-node-modules-${VERSION}.tar.gz"
+LOCAL_N="reaper-local-n-${VERSION}.tar.gz"
+
+tempDir=$(mktemp -d)
+
+function cleanup {
+	exit_code=$?
+	set +e
+	if [[ -n $tempDir ]];then
+		rm -Rf $tempDir
+		echo "Deleted $tempDir."
+	fi
+	exit $exit_code
+}
+
+trap cleanup EXIT SIGINT SIGTERM
+
+mkdir -p ${tempDir}/mariner_caches
+
+marinerCacheDir=${tempDir}/mariner_caches
+
+function checkInternet {
+	sudo tdnf install -y nc
+	nc -z 8.8.8.8 443
+	if [ $? -eq 0 ]; then
+		echo "Online."
+	else
+		echo "Offline, reaper needs internet connectivity to download and build cache."
+		exit 1
+	fi
+}
+
+function dieIfError {
+	if [ $? -ne 0 ];then
+		echo "Error $?, exiting build."
+		exit $?
+	fi
+}
+
+function installNodeModules {
+	echo "Installing node modules."
+	sudo tdnf install -y nodejs | dieIfError
+	npm config set cache "$tempDir/.npm" --global
+	# Default node/npm versions in Mariner fails to build dependency node module versions due to known
+	# incompatibilities.
+	# Backward compatible with node@v14.18.0
+	# When installing modules via npm to default prefix='/usr/local' in mariner, the permissions for 'others'
+	# is incoorectly set that causes 'which' to still point to older path, as access/newfstatat fail with -ENOPERM
+	# Setting a new global npm folder for fixing permission issues.
+	# (works well with id=0, but reaper build will fail.)
+	npm config set prefix "$HOME/.npm-global"
```

Comment:
Let's use a folder inside `tempDir` so we're sure it's clean and gets removed after we're done.

Reply:
using a custom folder inside tmp for all caches path.